ion/src/test/java/com/fasterxml/jackson/dataformat/ion/polymorphism/PolymorphicRoundTripWithSerializationAnnotations.java

```diff
@@ -0,0 +1,94 @@
+package com.fasterxml.jackson.dataformat.ion.polymorphism;
+
+import com.amazon.ion.IonValue;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.dataformat.ion.IonGenerator;
+import com.fasterxml.jackson.dataformat.ion.IonObjectMapper;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class PolymorphicRoundTripWithSerializationAnnotations {
+
+    private static final String SUBCLASS_TYPE_NAME =
+            PolymorphicRoundTripWithSerializationAnnotations.Subclass.class.getTypeName();
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
+    static public abstract class BaseClass { /* empty */ }
+
+    public static class Subclass extends BaseClass {
+        public String someString;
+        public int anInt;
+
+        public Subclass() {};
+        public Subclass(String s, int i) {
+            this.someString = s;
+            this.anInt = i;
+        }
+    }
+
+    @Test
+    public void testNativeTypeIdsEnabledOnWriteByDefault() throws IOException {
+        IonObjectMapper mapper = new IonObjectMapper();
+        Subclass subclass = new Subclass("some value", 42);
+        IonValue subclassAsIon = mapper.writeValueAsIonValue(subclass);
+        String expectedIonStr = "\n" +
+                "'" + SUBCLASS_TYPE_NAME + "'::{\n" +
+                "  someString:\"some value\",\n" +
+                "  anInt:42\n" +
+                "}";
```

Comment:
Comparing Ion data for string equality isn't a good idea because there isn't a canonical encoding. For example, the IonJava text writer (which is used in `IonValue.toPrettyString`) could legally start writing a space between field names and field values, and this test would start failing even though it is data model equivalent.

Therefore, I recommend testing data model equivalence by parsing your expected value into an IonValue and using IonJava's `Equivalence` utility class to perform the comparison.

(This comment applies to all tests added in this PR.)

Reply:
Thanks for the tip! I'll do that.